lib/portage/dbapi/bintree.py

```diff
@@ -493,47 +555,78 @@ def move_ent(self, mylist, repo_match=None):
 				writemsg("!!! "+mycpv+" -> "+mynewcpv+"\n", noiselevel=-1)
 				continue
 
-			tbz2path = self.getname(mycpv)
-			if os.path.exists(tbz2path) and not os.access(tbz2path,os.W_OK):
+			binpkg_path = self.getname(mycpv)
+			if os.path.exists(binpkg_path) and \
+				not os.access(binpkg_path,os.W_OK):
 				writemsg(_("!!! Cannot update readonly binary: %s\n") % mycpv,
 					noiselevel=-1)
 				continue
 
 			moves += 1
-			mytbz2 = portage.xpak.tbz2(tbz2path)
-			mydata = mytbz2.get_data()
+			binpkg_format = mycpv._metadata.get('BINPKG_FORMAT', None)
+			if not binpkg_format:
+				binpkg_format = get_binpkg_format(binpkg_path)
```

Comment:
Since `mycpv` is returned from `self.dbapi.cp_list`, we can assume that `mycpv._metadata['BINPKG_FORMAT']` has the correct `BINPKG_FORMAT` value here.

Reply:
I've addressed this in https://github.com/RinCat/portage/pull/2.